src/test/java/testsuite/regression/StatementRegressionTest.java

```diff
@@ -12707,4 +12707,44 @@ public void testBugIssue450() throws Exception {
             }
         }
     }
+
+    /**
+     * Test fix for Issue #464. Executing a batch statement should ignore any comments.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testBugIssue464() throws Exception {
+        Properties props = new Properties();
+        props.setProperty(PropertyKey.sslMode.getKeyName(), SslMode.DISABLED.name());
+        props.setProperty(PropertyKey.allowPublicKeyRetrieval.getKeyName(), "true");
+        this.conn = getConnectionWithProps(props);
+
+        String tableName = "testBugIssue464";
+
+        final Statement statement = this.conn.createStatement();
+        createTable(tableName, "(c0 INT)");
+
+        Statement bstmt = conn.createStatement();
+        bstmt.addBatch("INSERT INTO " + tableName + " /* Comment 1 */ VALUES (1)");
+        bstmt.addBatch("INSERT INTO " + tableName + " VALUES (2)" + " # Comment 2");
+        bstmt.addBatch("INSERT INTO " + tableName + " VALUES (3)" + " -- Comment 3");
+        bstmt.addBatch("/* Comment 4 */");
+        bstmt.addBatch("# Comment 5");
+        bstmt.addBatch("-- Comment 6");
+
+        try {
+            bstmt.executeBatch();
```

Comment:
is addBatch() that returns NONE or executeBatch() or both?

Reply:
The executeBatch() method triggers the query type check, addBatch() will not.